### PR TITLE
AMBARI-23456. Generate markdown docs from internal annotations.

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
- remove /properties endpoint
- generate markdown files based on annotations (for later usage - probably with mkdocs, but at least in README.md files on github)
- create docs folder (empty one) - do not fill those folders yet, i will try to create better examples for shipper configurations.
- generate docs with maven exec plugin + use it with make (i used prop-docs instead of docs - probably the docs will be with javadoc + prop-docs + mkdoc output)

## How was this patch tested?
manually in IDE and with maven

Please review @g-boros

### CODE REVIEW

**Comment by oleewere** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/97/
Test PASSed.

File: ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/LogSearchMarkdownGenerator.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class LogSearchMarkdownGenerator {
+
+  private static final String TOP_LEVEL_SECTIONS = "topLevelSections";
+  private static final String INPUT_CONFIG_SECTION = "inputConfigSection";
+  private static final String FILTER_CONFIG_SECTION = "filterConfigSection";
+  private static final String POST_MAP_VALUES_SECTION = "postMapValuesSection";
+  private static final String LOGSEARCH_PROPERTIES_TEMPLATE_KEY = "logsearchProperies";
```
Comment (adoroszlai) [inline ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/LogSearchMarkdownGenerator.java]:
```suggestion
  private static final String LOGSEARCH_PROPERTIES_TEMPLATE_KEY = "logsearchProperties";
```

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/99/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/100/
Test PASSed.

File: Makefile [hunk 2]
```diff
@@ -46,6 +46,9 @@ rpm:
 deb:
 	$(MAVEN_BINARY) clean package -Dbuild-deb -DskipTests -Djdk.version=$(LOGSEARCH_JAVA_VERSION)
 
+prop-docs:
+	$(MAVEN_BINARY) -pl ambari-logsearch-docs clean package exec:java -DskipTests -Djdk.version=$(LOGSEARCH_JAVA_VERSION)
```
[the remark this answers is not in the document]
Comment (oleewere) [reply Makefile]: @adoroszlai , i have added a prop-docs-only as well without install dependency, as building the web package is really slow

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/101/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/102/
Test PASSed.

File: ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/ShipperConfigDesritionDataHolder.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.ambari.logsearch.doc;
+
+import java.util.List;
+
+public class ShipperConfigDesritionDataHolder {
```
Comment (adoroszlai) [inline ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/ShipperConfigDesritionDataHolder.java]:
```suggestion
public class ShipperConfigDescriptionDataHolder {
```

File: ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/ShipperConfigDescritionDataHolder.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.ambari.logsearch.doc;
+
+import java.util.List;
+
+public class ShipperConfigDescritionDataHolder {
```
Comment (adoroszlai) [inline ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/ShipperConfigDescritionDataHolder.java]: Sorry, but it's still not right.

```suggestion
public class ShipperConfigDescriptionDataHolder {
```

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/103/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/104/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/105/
Test PASSed.

**Pull request merged**